.baseline/checkstyle/checkstyle.xml

```diff
@@ -284,6 +284,10 @@
             <property name="format" value="^[a-z][a-zA-Z0-9]+$"/>
             <message key="name.invalidPattern" value="Member name ''{0}'' must match pattern ''{1}''."/>
         </module>
+        <module name="ConstantName">
+            <property name="format" value="^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"/>
```

Comment:
you can use possessive quantifiers. That way the matching might be simpler.
I didn't benchmark it, but checksyle can take considerable time overall, and small things like this add up.

Reply:
I updated the regexp from
`^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$`
to
`^[A-Z][A-Z0-9]+(_[A-Z0-9]+)*$`
Pls mind the `*` vs `+` char at pos 15. Pls clarify whether or not this is what you were pointing out.

Reply:
**Update:** I reverted this change as it breaks for `Z_COLUMN`, `Z_SCHEMA` and `Z_SORT_ORDER`.

Pls point out if you suggest a different regexp here. 🤔

Reply:
to leverage possessive quantifiers replace 
```
^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$
```

with 
```
^[A-Z][A-Z0-9]*+(_[A-Z0-9]++)*+$
```

Reply:
Thanks, I see. I experimented a bit with the possessive quantifiers in the regexes as well as removing some custom messages that seem redundant. 

I created a draft PR with these changes: #10681. Even though it seems like just a few lines of changes, it has an impact for the whole codebase and I don't want to hold up this one. (I'd prefer to keep the scope of this PR tight, really just renaming the constants to uppercase and enforcing that rule.)

Reply:
I think for this PR we should just stick with the [default regex pattern of checkstyle](https://checkstyle.sourceforge.io/checks/naming/constantname.html), which I believe is what you had initially. Improving the regex can be done in a separate PR.

Reply:
@nastra, yep, same thought here, hence the spin-off draft PR.

In this PR the definition I used is this (this is the current state right now):
`<property name="format" value="^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"/>`

the default you linked is this:
`<property name="format" ="^log(ger)?$|^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"/>`

So the only difference is I removed the first part the allows for the lowercase `log(ger)` as this codebase doesn't use it, so it was redundant.

Also we have the custom message in place to be consistent with the rest of the definitions for now.

Pls LMK if this is all good as it right now or we need some changes here.